packages/aws-cdk-lib/aws-appconfig/lib/environment.ts

```diff
@@ -116,6 +116,22 @@ abstract class EnvironmentBase extends Resource implements IEnvironment, IExtens
   public addExtension(extension: IExtension) {
     this.extensible.addExtension(extension);
   }
+
+  public grant(grantee: iam.IGrantable, ...actions: string[]) {
+    return iam.Grant.addToPrincipal({
+      grantee,
+      actions,
+      resourceArns: [`${this.environmentArn}/*`],
```

Comment:
I'm not too familiar with appconfig usage. Any document supporting that this is the correct arns and least privilege?

Reply:
Thank you for your comment.

Upon reviewing it again, I found that the scope of the `resourcesArn` for `grantReadConfig` was too broad, so I made a modification (added `/configuration/`).

In AppConfig, a Configuration Profile represents a parameter (e.g., a Feature Flag), and an Environment represents a logical group of deployed Configuration Profiles.  ([docs](https://docs.aws.amazon.com/appconfig/latest/userguide/what-is-appconfig.html#learn-more-appconfig-how-it-works))

The ARN of a Configuration belonging to an Environment takes the following format.  ([docs](https://docs.aws.amazon.com/service-authorization/latest/reference/list_awsappconfig.html#awsappconfig-resources-for-iam-policies))

> arn:${Partition}:appconfig:${Region}:${Account}:application/${ApplicationId}/environment/${EnvironmentId}/configuration/${ConfigurationProfileId}

For this issue, I wanted to grant read permission for the parameters (Configurations) belonging to a logical group (Environment), so I needed to use a wildcard for `/configuration/ `and beyond, as I realized.

Previously, the wildcard was applied from `/${EnvironmentId}/` onward, which was too broad in scope.

Regarding the `grant` method, I changed it to limit the resources to the Environment only.